include/gazebo_mavlink_interface.h

```diff
@@ -248,8 +251,8 @@ class GazeboMavlinkInterface : public ModelPlugin {
   void ImuCallback(ImuPtr& imu_msg);
   void GpsCallback(GpsPtr& gps_msg);
   void GroundtruthCallback(GtPtr& groundtruth_msg);
-  void LidarCallback(LidarPtr& lidar_msg);
-  void SonarCallback(SonarPtr& sonar_msg);
+  void LidarCallback(LidarPtr& lidar_msg, const int& id);
+  void SonarCallback(SonarPtr& sonar_msg, const int& id);
```

Comment:
My taste would be to pass the `shared_ptr` by value and the int as well instead of reference but probably both work.

Reply:
It does work right. And the `shared_ptr` was already passed this way on the callbacks (it's actually a plugin convention so I think it's right).